iambic/plugins/v0_1_0/aws/handlers.py

```diff
@@ -165,6 +165,10 @@ async def apply_iam_templates(
     :param templates: The list of templates to apply.
     :param remote_worker: The remote worker to use for applying templates.
     """
+    if config.spoke_role_is_read_only:
+        log.critical("Unable to apply resources when spoke_role_is_read_only is True")
+        return []
```

Comment:
should we raise an error instead? I guess we don't want to raise because we don't want to interrupt the other plugins?

Reply:
yeah I didn't want AWS to interrupt execution of other providers. Ideally we should return template change details with the exception in each instance but the way we output proposed changes will need to be updated and didn't want the ticket to go down a rabbit hole.

Reply:
yea, its an aspect of provider isolation in `apply`. because if okta has a service outage, I still want AWS to work.